pkg/gcp/machine_controller_util.go

```diff
@@ -83,34 +83,32 @@ func (ms *MachinePlugin) CreateMachineUtil(ctx context.Context, machineName stri
 		instance.Description = *providerSpec.Description
 	}
 
-	disks := []*compute.AttachedDisk{}
+	var disks []*compute.AttachedDisk
 	for _, disk := range providerSpec.Disks {
-		var attachedDisk compute.AttachedDisk
-		autoDelete := false
-		if disk.AutoDelete == nil || *disk.AutoDelete == true {
-			autoDelete = true
+		attachedDisk := compute.AttachedDisk{
+			Type:       validation.DiskTypePersistent,
+			Boot:       disk.Boot,
+			AutoDelete: false,
```

Comment:
Maybe `AutoDelete: ptr.Deref(disk.AutoDelete, false)` and delete the other code below that does this

Reply:
Good suggestion, but shouldn't it be then `AutoDelete: ptr.Deref(disk.AutoDelete, true)`?